trie/trienode/node.go

```diff
@@ -186,9 +206,9 @@ func NewWithNodeSet(set *NodeSet) *MergedNodeSet {
 // Merge merges the provided dirty nodes of a trie into the set. The assumption
 // is held that no duplicated set belonging to the same trie will be merged twice.
 func (set *MergedNodeSet) Merge(other *NodeSet) error {
-	_, present := set.Sets[other.Owner]
+	subset, present := set.Sets[other.Owner]
 	if present {
-		return fmt.Errorf("duplicate trie for owner %#x", other.Owner)
+		return subset.Merge(other.Owner, other.Nodes)
```

Comment:
hm Could u explain why they change the expected behaviour here?

Reply:
They did not explicitly explain that here. 
My intuition is that, although the current flow will never get to this error, the merge function should remain a merge function, that is, to merge NodeSets. Any unexpected behaviors should be from somewhere else and thus should be handled at those places. 
Therefore, it seems reasonable here. What do you think?

Reply:
The new flow, should be accept the dirtied nodes change then overwrite it into nodeset with same owner with update flag tracking counter too, reject in case viloate owner (some issues here) It could be a general solution. Actually we don't run previous logic in real prod yet, so we can't make sure that `fmt.Errorf("duplicate trie for owner %#x", other.Owner)` can happen in real prod or not. But from my view, the new logic is quite cleary.